asset/asset.go

```diff
@@ -1182,12 +1219,29 @@ func (g *GroupKeyRevealTapscript) Validate(assetID ID) error {
 	return nil
 }
 
+// GkrV1Version is the version of the group key reveal.
+//
+// Version 1 is the original version that's based on an OP_RETURN.
+//
+// Version 2 is a follow up version that instead uses a Pedersen commitment.
+type GkrV1Version = uint8
```

Comment:
The type name is confusing to me in two different ways here: it conflates with "version 1" of the group key reveal, and that there's `V1` in the name.

I think we should name this `NonSpendLeafVersion`. There's no need to tie this version type (or non-sepnd leaf fucntionality) directly to Group Key Reveal V1, as the non-spend leaf version is downstream of GKR V1. In fact, GKR depends on the non-spend leaf version, not the other way around.

Reply:
SGTM, swapped to be `NonSpendLeafVersion`. It's just a `uint8` on the wire, so in the future, we can update this constant name if we need to.